libp2p/Host.cpp

```diff
@@ -196,6 +196,25 @@ void Host::stopCapabilities()
     }
 }
 
+std::shared_ptr<Peer> Host::peer(NodeID const& _n) const
+{
+    RecursiveGuard l(x_sessions);
+    auto it = m_peers.find(_n);
+    if (it == m_peers.end())
+    {
+        LOG(m_logger) << "Peer " << _n << " not found";
+        return nullptr;
+    }
+    return it->second;
+}
+
+void Host::handshakeFailed(NodeID const& _n, HandshakeFailureReason _r)
+{
+    std::shared_ptr<Peer> p = peer(_n);
+    assert(p);
```

Comment:
This I think will fail when it's the first handshake attempt that failed and `m_peers` don't have the peer yet (for incoming connections it's added only after successful handshake)
No need to set it to the peer in that case.

Reply:
> 
> 
> This I think will fail when it's the first handshake attempt that failed and `m_peers` don't have the peer yet (for incoming connections it's added only after successful handshake)
> No need to set it to the peer in that case.

I don't think this case is possible - for outgoing connections we always decide which node to connect to using m_peers and the following check in RLPXHandshake protects against the failed incoming connection case: https://github.com/ethereum/aleth/blob/f1f5db331188cafe11db27f03b907b1475639742/libp2p/RLPxHandshake.cpp#L262-L263

Reply:
@gumb0 : Actually, the EIP8 tests hit this so I've replaced the assert with a nullptr check and log an error now (with VerbosityError since we never expect to hit this when running Aleth, just during tests, so if we see the error message there's a bug and we should investigate)